### DISCUSSION TITLE
Is it possible to access xrdp through a proxy server?

### DISCUSSION THREAD

**Discussion by matt-vonhippel** [neutrinolabs/xrdp#2828] [Q&A]
I am attempting to set up RDP access to an Ubuntu desktop from a Windows laptop. The desktop is only accessible through a proxy server, for which I have login information. I am able to SSH using this proxy from the Windows laptop, via the proxy server, to the Ubuntu desktop, but I can't figure out how to use RDP over the same pathway. Advice I've found online for a Windows RDP server suggests that one needs to set up something called an RDP Gateway Server. Is there something analogous to that for xrdp? Or is there another way to accomplish this?

**Comment by matt335672** [comment]
What kind of proxy do you have?

**Comment by matt-vonhippel** [comment]
I'm not sure, in part because I'm not enough of an expert to know what the different kinds of proxy are and in part because the proxy server was set up by my employer. If routing RDP through this proxy would require reconfiguring the proxy server itself then it's something I'd have to ask our scientific computing guy about, I just wanted to see if this was something I could do on my own, since they had already set up SSH access.

**Comment by matt335672** [comment]
RDP Gateway server is a separate product which provides a proxy for RDP connections. It would be separate to your existing proxy.

If your proxy allows you to make a straight connection from the Windows machine to the Ubuntu machine you can look at tunnelling the RDP connection over ssh using the '-L' option.

The idea is to forward connections to `127.0.0.2:3389` on the Windows machine to `localhost:3389` on the Ubuntu machine. Here's how the setup looks if you're using PuTTY as an ssh client:-
![Capture](https://github.com/neutrinolabs/xrdp/assets/30179339/6f9bbf3f-2ba4-48e3-aaca-33b2a5b6f887)

With that in place you connect to the Ubuntu machine over ssh, and while the ssh connection is active, start an RDP connection to 127.0.0.2. ssh does the rest.

Does it make sense, and could you try that? Come back to me if it doesn't make sense - I've left out some of the detail.
